dm/loader/checkpoint.go

```diff
@@ -528,34 +552,61 @@ func (cp *LightningCheckpointList) RegisterCheckPoint(ctx context.Context, worke
+func (cp *LightningCheckpointList) UpdateStatus(ctx context.Context, status lightingLoadStatus) error {
 	connection, err := cp.db.GetBaseConn(ctx)
 	if err != nil {
 		return terror.WithScope(terror.Annotate(err, "initialize connection"), terror.ScopeDownstream)
 	}
+	defer conn.CloseBaseConnWithoutErr(cp.db, connection)
 
+	sql := fmt.Sprintf("UPDATE %s set status = ? WHERE `task_name` = ? AND `source_name` = ?", cp.tableName)
+	cp.logger.Info("update lightning loader status",
+		zap.String("task", cp.taskName), zap.String("source", cp.sourceName),
+		zap.Stringer("status", status))
 	tctx := tcontext.NewContext(ctx, log.With(zap.String("job", "lightning-checkpoint")))
-	query := fmt.Sprintf("SELECT `worker_name` from %s where `task_name`=?", cp.tableName)
-	rows, err := connection.QuerySQL(tctx, query, taskName)
+	_, err = connection.ExecuteSQL(tctx, nil, "lightning-checkpoint", []string{sql},
+		[]interface{}{status, cp.taskName, cp.sourceName})
 	if err != nil {
-		return terror.WithScope(err, terror.ScopeDownstream)
+		return terror.WithScope(terror.Annotate(err, "update lightning status"), terror.ScopeDownstream)
+	}
+	return nil
+}
+
+func (cp *LightningCheckpointList) taskStatus(ctx context.Context) (lightingLoadStatus, error) {
+	connection, err := cp.db.GetBaseConn(ctx)
+	if err != nil {
+		return lightningStatusInit, terror.WithScope(terror.Annotate(err, "initialize connection"), terror.ScopeDownstream)
+	}
+	defer conn.CloseBaseConnWithoutErr(cp.db, connection)
+
+	query := fmt.Sprintf("SELECT status FROM %s WHERE `task_name` = ? AND `source_name` = ?", cp.tableName)
+	tctx := tcontext.NewContext(ctx, log.With(zap.String("job", "lightning-checkpoint")))
+	rows, err := connection.QuerySQL(tctx, query, cp.taskName, cp.sourceName)
+	if err != nil {
+		return lightningStatusInit, err
 	}
 	defer rows.Close()
-	var workerName string
-	for rows.Next() {
-		err = rows.Scan(&workerName)
-		if err != nil {
-			return terror.WithScope(terror.DBErrorAdapt(err, terror.ErrDBDriverError), terror.ScopeDownstream)
-		}
-		cpdb := config.TiDBLightningCheckpointPrefix + dbutil.TableName(workerName, taskName)
-		sql := fmt.Sprintf("DROP DATABASE IF NOT EXISTS %s", cpdb)
-		_, err = connection.ExecuteSQL(tctx, nil, "lightning-checkpoint", []string{sql})
-		if err != nil {
-			return terror.WithScope(err, terror.ScopeDownstream)
+	if rows.Next() {
+		var status lightingLoadStatus
+		if err = rows.Scan(&status); err != nil {
+			return lightningStatusInit, terror.WithScope(err, terror.ScopeDownstream)
 		}
+		return status, nil
 	}
-	query = fmt.Sprintf("DELETE from %s where `task_name`=?", cp.tableName)
-	_, err = connection.ExecuteSQL(tctx, nil, "lightning-checkpoint", []string{query}, []interface{}{taskName})
+	// status row doesn't exist, return default value
+	return lightningStatusInit, nil
+}
+
+func (cp *LightningCheckpointList) RemoveTaskCheckPoint(ctx context.Context) error {
```

Comment:
when is this function to be called ?

Reply:
No used no. It is orginally designed for being calling at cleanup meta, but the current design just call raw `drop table` like other tables. 😅

Reply:
so we can delete this func?